### PR TITLE
[stdlib] Changed methods in Mirror to computed properties

### PR DESCRIPTION
**Pull request by harlanhaskins**
Also fixes a bug when resolving a pointer to a function that uses a computed property.

### CODE REVIEW

**Comment by harlanhaskins** [comment]
Oops, forgot to make a separate commit for the print overloads. I'm excited to get rid of these once generics work.

**Review by segiddins — APPROVED** [review]
Looks great! Let's rebase and see this pass on CI?

**Comment by harlanhaskins** [comment]
Will do now.

**Comment by harlanhaskins** [comment]
Whoops, I failed one of the tests! 😃

File: examples/protocol.tr [hunk 1]
```diff
@@ -8,7 +8,7 @@ type Bar: Debuggable {
   }
 
   func debugDescription() -> String {
-    return String(cString: Mirror(reflecting: self).name())
+    return String(cString: Mirror(reflecting: self).name)
```
Comment (segiddins) [inline examples/protocol.tr]: `typeName`?

**Comment by segiddins** [comment]
It's passing, :ship:

**Pull request merged**